### PR TITLE
Add public interface functions to NSFVAction

### LINKED ISSUE #24674: Add more public interface functions to NSFVAction
## Reason

I have implemented a domain overlapping coupling between Pronghorn and SAM.
The approach uses the NSFVAction to avoid input duplication.
But to use the NSFVAction I need a few public interface function.

## Design

Added public interface functions.

## Impact

None.

### PR DESCRIPTION
**Pull request by snschune**
close #24674 

## Reason

Needed to use info in NSFVAction in another action.

## Design

Added getters to relevant data.

## Impact

none expected

### CODE REVIEW

File: modules/navier_stokes/include/base/NSFVBase.h [hunk 1]
```diff
@@ -3508,3 +3536,28 @@ NSFVBase<BaseType>::checkRhieChowFunctorsDefined()
     paramError("add_flow_equations",
                "Rhie Chow coefficient az must be provided for advection by auxiliary velocities");
 }
+
+template <class BaseType>
+std::string
+NSFVBase<BaseType>::velocityName(unsigned int j) const
+{
+  if (j >= _velocity_name.size())
+    mooseError("Argument j = ", j, " out of bounds");
```
Comment (GiudGiud) [inline modules/navier_stokes/include/base/NSFVBase.h:3545]:
```suggestion
    mooseError("Dimension j = ", j, " does not exist");
```

File: modules/navier_stokes/include/base/NSFVBase.h [hunk 2]
```diff
@@ -26,6 +26,34 @@ class NSFVBase : public BaseType
...
+  std::string rhieChowName() const
+  {
+    if (_porous_medium_treatment)
+      return "pins_rhie_chow_interpolator";
+    return "ins_rhie_chow_interpolator";
+  }
+  ///@}
+
+  ///@{ public interface for variable and property names
+  std::string pressureName() const { return _pressure_name; }
```
Comment (GiudGiud) [inline modules/navier_stokes/include/base/NSFVBase.h:42]: use the specific type (NonLinearVariableName iirc)

Comment (lindsayad) [reply modules/navier_stokes/include/base/NSFVBase.h:42]:
```suggestion
  NonlinearVariableName pressureName() const { return _pressure_name; }
```

File: modules/navier_stokes/include/base/NSFVBase.h [hunk 3]
```diff
@@ -26,6 +26,34 @@ class NSFVBase : public BaseType
...
+  ///@{ general public interface functions
+  bool hasEnergyEquation() const { return _has_energy_equation; }
+  bool hasScalar() const { return _has_scalar_equation; }
+  unsigned int dim() const { return _dim; }
+  std::string rhieChowName() const
+  {
+    if (_porous_medium_treatment)
+      return "pins_rhie_chow_interpolator";
+    return "ins_rhie_chow_interpolator";
+  }
```
Comment (lindsayad) [inline modules/navier_stokes/include/base/NSFVBase.h:38]: It would be nice to split multi-line definitions out of the class definition.

Comment (lindsayad) [reply modules/navier_stokes/include/base/NSFVBase.h:38]:
```suggestion
  std::string rhieChowName() const
  {
    if (_porous_medium_treatment)
      return prefix() + "pins_rhie_chow_interpolator";
    return prefix() + "ins_rhie_chow_interpolator";
  }
```

File: modules/navier_stokes/include/base/NSFVBase.h [hunk 4]
```diff
@@ -26,6 +26,34 @@ class NSFVBase : public BaseType
...
+  ///@}
+
+  ///@{ public interface for variable and property names
+  std::string pressureName() const { return _pressure_name; }
+  std::string fluidTemperatureName() const { return _fluid_temperature_name; }
```
Comment (lindsayad) [inline modules/navier_stokes/include/base/NSFVBase.h:43]:
```suggestion
  const NonlinearVariableName & fluidTemperatureName() const { return _fluid_temperature_name; }
```

File: modules/navier_stokes/include/base/NSFVBase.h [hunk 5]
```diff
@@ -26,6 +26,34 @@ class NSFVBase : public BaseType
...
+  std::string velocityName(unsigned int j) const;
+  std::vector<std::string> velocityNames() const { return _velocity_name; }
+  std::vector<NonlinearVariableName> passiveScalarNames() const { return _passive_scalar_names; }
```
Comment (lindsayad) [inline modules/navier_stokes/include/base/NSFVBase.h:46]:
```suggestion
  const std::string & velocityName(unsigned int j) const;
  const std::vector<std::string> & velocityNames() const { return _velocity_name; }
  const std::vector<NonlinearVariableName> & passiveScalarNames() const { return _passive_scalar_names; }
```

File: modules/navier_stokes/include/base/NSFVBase.h [hunk 6]
```diff
@@ -3508,3 +3536,28 @@ NSFVBase<BaseType>::checkRhieChowFunctorsDefined()
     paramError("add_flow_equations",
                "Rhie Chow coefficient az must be provided for advection by auxiliary velocities");
 }
+
+template <class BaseType>
+std::string
+NSFVBase<BaseType>::velocityName(unsigned int j) const
```
Comment (lindsayad) [inline modules/navier_stokes/include/base/NSFVBase.h:3542]:
```suggestion
const std::string &
NSFVBase<BaseType>::velocityName(const unsigned int dim) const
```

File: modules/navier_stokes/include/base/NSFVBase.h [hunk 7]
```diff
@@ -3508,3 +3536,28 @@ NSFVBase<BaseType>::checkRhieChowFunctorsDefined()
...
+{
+  if (j >= _velocity_name.size())
```
Comment (lindsayad) [inline modules/navier_stokes/include/base/NSFVBase.h:3544]:
```suggestion
  if (dim >= _velocity_name.size())
```

File: modules/navier_stokes/include/base/NSFVBase.h (hunk 1 above)
Comment (lindsayad) [reply modules/navier_stokes/include/base/NSFVBase.h:3545]:
```suggestion
    mooseError("Argument dim = ", dim, " out of bounds");
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Review by lindsayad — APPROVED** [review]

**Comment by lindsayad** [comment]
In the future @snschune let's try to batch those single line changes into less commits. Thank you for all these contributions though!

**Pull request merged**